graph/tests/boost-examples.rkt

```diff
@@ -135,6 +135,14 @@
 (check-false (eq? g/for-copy g/copy))
 (check-equal? g/for-copy g/copy)
 
+;; graph union
+(define g/for-union
+  (undirected-graph '((a c) (a d) (b a) (b d))))
+(define g/for-union2
+  (undirected-graph '((c f) (d c) (d e) (d f) (e b) (e g) (f e) (f g))))
+(graph-union! g/for-union g/for-union2)
+(check-equal? g/for-union g/for-union)
```

Comment:
Sorry to nit, but won't `check-equal`ing the same two things always pass?  I don't see how this tests the `graph-union` operation.